nmtrust

```diff
@@ -45,6 +45,19 @@ file_check() {
     fi
 }
 
+check_connection() {
+    local name=$1
+    local connection_excluded=false
+    local excludes=($(grep -v '^#' < $EXCLUDEFILE))
```

Comment:
Here you are reading the contents of the file in as a string, and then asking the shell to try to split it into an array however it sees fit. I think it is [considered safer](https://github.com/koalaman/shellcheck/wiki/SC2207) if the file is explicitly read in as an array with one element per line using `mapfile`:

    mapfile -t excludes < <(grep -v '^#' < $EXCLUDEFILE)

Does the wildcard matching still work for you if you do it this way?

Reply:
Yes, this works as well. Thanks for the hint.

Reply:
Did you check the PS with Shellcheck? I heard about this before, but didn't try it yet...

Reply:
Yeah, I find Shellcheck useful guidance for my haphazard Bash skills.